Packs/Base/Scripts/DBotMLFetchData/DBotMLFetchData.py

```diff
@@ -1,6 +1,1031 @@
+    for t in HTML_TAGS:
+        html_counter[t] = 0 if t not in html_counter else html_counter[t]
+    html_counter = {k: v for k, v in html_counter.items() if k in HTML_TAGS}
+    return html_counter
+
+
+def get_ml_features(tokenized_text):
+    global EMBEDDING_DICT
+    if EMBEDDING_DICT is None:
+        with open('/var/glove_top_20k.p', 'rb') as file:
+            EMBEDDING_DICT = pickle.load(file)
+    vectors = [EMBEDDING_DICT[w] for w in tokenized_text if w in EMBEDDING_DICT]
+    if len(vectors) == 0:
+        mean_vector = np.zeros(50)
+    else:
+        mean_vector = np.mean(vectors, axis=0)
+    res = {'glove_avg_{}'.format(i): mean_vector[i].item() for i in range(len(mean_vector))}
+    return res
+
+
+def get_header_value(email_headers, header_name, index=0, ignore_case=False):
+    if ignore_case:
+        header_name = header_name.lower()
+        headers_with_name = [header_dict for header_dict in email_headers if header_dict['name'].lower() == header_name]
+    else:
+        headers_with_name = [header_dict for header_dict in email_headers if header_dict['name'] == header_name]
+    if len(headers_with_name) == 0:
+        return None
+    else:
+        return headers_with_name[index]['value']
+
+
+def get_headers_features(email_headers):
+    spf_result = get_header_value(email_headers, header_name='Received-SPF')
+    if spf_result is not None:
+        spf_result = spf_result.split()[0].lower()
+    else:
+        spf_result = 'other'
+    res_spf = {k: 0 for k in ['none', 'neutral', 'pass', 'fail', 'softfail', 'other', 'temperror', 'permerror']}
+    if spf_result in res_spf:
+        res_spf[spf_result] += 1
+    else:
+        res_spf['other'] += 1
+    res_spf['non-positive'] = spf_result in set(['none', 'fail', 'softfail'])
+    authentication_results = get_header_value(email_headers, header_name='Authentication-Results')
+    if authentication_results is not None:
+        if 'dkim=' in authentication_results:
+            dkim_result = authentication_results.split('dkim=')[-1].split()[0].lower().strip()
+        else:
+            dkim_result = 'other'
+    else:
+        dkim_result = 'other'
+    res_dkim = {k: 0 for k in ['none', 'neutral', 'pass', 'fail', 'softfail', 'other', 'temperror', 'permerror']}
+    if dkim_result in res_dkim:
+        res_dkim[dkim_result] += 1
+    else:
+        res_dkim['other'] += 1
+    res_dkim['non-positive'] = spf_result in set(['none', 'fail', 'softfail'])
+    list_unsubscribe = get_header_value(email_headers, header_name='List-Unsubscribe')
+    unsubscribe_post = get_header_value(email_headers, header_name='List-Unsubscribe-Post-SPF')
+    res_unsubscribe = 1 if list_unsubscribe is not None or unsubscribe_post is not None else 0
+    from_value = get_header_value(email_headers, header_name='From')
+    return_path_value = get_header_value(email_headers, header_name='Return-Path')
+    last_received_value = get_header_value(email_headers, header_name='Received', index=-1)
```

Comment:
add also "reply-to" header and compare to the from address (I think envelope sender represent by the return path so you already extract that)
also maybe worth to extract the domain  for each one to compare between them